### PR TITLE
Create the k3s package using the upstream installer script

### PR DESCRIPTION
**Pull request by jimmykarily**
Hoping to replace [these commands](https://github.com/kairos-io/provider-kairos/blob/9f56a5578bfee5d0d0018d4eba7686a19939f7fb/Earthfile#L168-L172) with a simple luet install.

### CODE REVIEW

**Review by mudler — APPROVED** [review]

**Comment by jimmykarily** [comment]
CI fails. It's reproducible locally with:

```
sudo luet build --tree packages k8s/k3s
```
not sure what's wrong yet.

**Comment by Itxaka** [comment]
` INFO   Error response from daemon: dockerfile parse error line 15: unknown instruction: EXPORT` ???

File: packages/k8s/k3s/build.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  category: "development"
+  version: ">=0"
+env:
+  - INSTALL_K3S_BIN_DIR="/usr/bin"
+  - INSTALL_K3S_SELINUX_WARN=true
+  - INSTALL_K3S_SKIP_START="true"
+  - INSTALL_K3S_SKIP_ENABLE="true"
+  - INSTALL_K3S_SKIP_SELINUX_RPM="true"
+prelude:
+{{$arch:=(default .Values.arch "amd64")}}
```
Comment (Itxaka) [inline packages/k8s/k3s/build.yaml:12]: could this be the issue? After all this is a templating instruction so the prelude will be empy

**Review by Itxaka — APPROVED** [review]
nice

**Comment by jimmykarily** [comment]
The steps are run as dockerfile `RUN` entries. I moved the setting of env vars in the `env:` section and fixed that issue. Not there is another problem, that the upstream installer script cannot detect systemd/openrc inside a docker image:

```
       #9 [5/9] RUN bash installer.sh
 INFO   #9 0.297 [ERROR]  Can not find systemd or openrc to use as a process supervisor for k3s
       #9 ERROR: process "/bin/sh -c bash installer.sh" did not complete successfully: exit code: 1
       ------
        > [5/9] RUN bash installer.sh:
       0.297 [ERROR]  Can not find systemd or openrc to use as a process supervisor for k3s

```

**Comment by Itxaka** [comment]
> The steps are run as dockerfile `RUN` entries. I moved the setting of env vars in the `env:` section and fixed that issue. Not there is another problem, that the upstream installer script cannot detect systemd/openrc inside a docker image:
> 
> ```
>        #9 [5/9] RUN bash installer.sh
>  INFO   #9 0.297 [ERROR]  Can not find systemd or openrc to use as a process supervisor for k3s
>        #9 ERROR: process "/bin/sh -c bash installer.sh" did not complete successfully: exit code: 1
>        ------
>         > [5/9] RUN bash installer.sh:
>        0.297 [ERROR]  Can not find systemd or openrc to use as a process supervisor for k3s
> ```

Im guessing you can get it to detect systemd by setting the `image` key as it was before so the base system has systemd or openrc or whatever.


Now Im guessing that is why the check for init.systemd was in there lol.

But it could also mean that we need two packages? one for systemd and one for openrc?

**Comment by jimmykarily** [comment]
For now I opted for installing service files for both systemd (openrc and systemd). No OS would ever consume both so just having the other one laying around shouldn't be a problem (or is it?)

**Comment by jimmykarily** [comment]
This is my ugly hack: https://github.com/kairos-io/packages/pull/335/files#diff-c9743e80f8388cd7b60efc400366bd8ef48a50f64f8c1e70bdb9c085d2f9cb4bR20-R30

File: packages/k8s/k3s/build.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  {{ if eq $arch "arm" }}
+  - ARCH=arm64
+  {{ else }}
+  - ARCH={{ $arch }}
+  {{ end }}
+steps:
+  - curl -sfL https://get.k3s.io > installer.sh
+  # Let the installer script install service files both for openrc and systemd:
+  # https://github.com/k3s-io/k3s/blob/36645e7311e9bdbbf2adb79ecd8bd68556bc86f6/install.sh#L114-L122
+  - touch /bin/systemctl && chmod +x /bin/systemctl
```
Comment (Itxaka) [inline packages/k8s/k3s/build.yaml:22]: :rofl:

**Review by Itxaka — APPROVED** [review]
if it works, it works!

File: packages/k8s/k3s/build.yaml (hunk 2 above)
Comment (jimmykarily) [reply packages/k8s/k3s/build.yaml:22]: Poor checks need poor hacks :p

File: packages/utils/k3d/build.yaml [hunk 3]
```diff
@@ -3,8 +3,7 @@ requires:
   category: "development"
   version: ">=0"
 prelude:
-{{$arch:=(default .Values.arch "amd64")}}
-
+{{$arch:=(default "amd64" .Values.arch)}}
```
Comment (jimmykarily) [inline packages/utils/k3d/build.yaml:6]: This is Spring right? http://masterminds.github.io/sprig/defaults.html

Comment (Itxaka) [reply packages/utils/k3d/build.yaml:6]: yup! https://github.com/mudler/luet/blob/master/go.mod#L6

**Pull request merged**